lms/djangoapps/certificates/views/webview.py

```diff
@@ -332,8 +333,14 @@ def _get_user_certificate(request, user, course_key, course, preview_mode=None):
     if preview_mode:
         # certificate is being previewed from studio
         if request.user.has_perm(PREVIEW_CERTIFICATES, course):
-            if course.certificate_available_date and not course.self_paced:
+            if (
+                course.certificates_display_behavior == CertificatesDisplayBehaviors.END_WITH_DATE
+                and course.certificate_available_date
+                and not course.self_paced
+            ):
                 modified_date = course.certificate_available_date
+            elif course.certificates_display_behavior == CertificatesDisplayBehaviors.END:
```

Comment:
@Tj-Tracy Do we need to make a change to the date we send to Credentials with this change?